src/language-css/parser-postcss.js

```diff
@@ -288,6 +288,13 @@ function parseNestedCSS(node) {
       node.raws.selector = selector;
     }
 
+    // https://github.com/postcss/postcss-custom-selectors/blob/495d9f3821257b10cd080ed2d1df81bc6db128a2/lib/custom-selectors-from-root.js#L34
+    // postcss-less@2.0.0 parse `custom-selector` as `css-decl`
+    if (node.type === "css-decl" && node.prop === "@custom-selector") {
+      selector = node.value;
+      node.raws.value = selector;
+    }
```

Comment:
After testing i find what you can't use `@custom-selector` (postcss plugin) with less, because less use `@var` notation for variables, using this potentially can break a `@custom-selector` variable in less code, it is the very rare case, but i think we should do not break code in all cases, so i think we should remove this.

what test is broken if we remove this?

Reply:
```less
@custom-selector :--camelCase .my-css-selector;
                ^
```

this will format into 

```less
@custom-selector: --camelCase .my-css-selector;
                ^
```

https://github.com/prettier/prettier/blob/master/tests/css_attribute/custom-selector.css

Reply:
also if we don't do this, instead of `postcss-selector-parser` we will parse it as value , hard to do something in print

Reply:
@fisker i think for Less we should use `postcss-values-parser` for `@custom-selector`, because `@custom-selector` in Less is just variable, you can use `@custom-selector` at-rule only in pure CSS.

Reply:
Oh, i found the problem, we use `less` parser by default for pure CSS https://github.com/prettier/prettier/blob/master/src/language-css/parser-postcss.js#L524, we should fix it too, we need use pure CSS (from `postcss`) parser for `css` extension

Reply:
Are you sure this is resolved? I guess it at least makes sense to add a check like `options.parser === 'css'` to that code for `@custom-selector`.

Reply:
Good solution for this PR, anyway we need migrate on standard css parser (built-in in postcss) for pure CSS, because less syntax is **incompatible** with pure CSS syntax

Reply:
so, add `options.parser === 'css'` ?